src/Microsoft.Health.Fhir.Core/Features/Search/SearchService.cs

```diff
@@ -22,7 +22,6 @@ namespace Microsoft.Health.Fhir.Core.Features.Search
     /// </summary>
     public abstract class SearchService : ISearchService
     {
-        private readonly ISearchOptionsFactory _searchOptionsFactory;
```

Comment:
>ISearchOptionsFactory [](http://example.com/codeflow?start=25&length=21)

why?

Reply:
And that's what I don't like.
Purpose of searchOptionFactory is to build searchOptions object which has Expression and other things.

And later we treat Expression based on persistence implementation.

request->common representation as expressions-> Cosmos or Sql logic which tailor expressions to it's own needs.

If we add Or condition (cosmos or sql) on second step, it's gonna be nightmare to support.

Reply:
Got the point. The OR condition in expressions are now removed. Instead, it records phases in continuation token and get the results by multiple searches.